### PR TITLE
Chat History

### PR DESCRIPTION
**Pull request by peterkeppert**
Draft PR for #52 to get an idea what needs to be done for basic functionality.

### CODE REVIEW

File: src/streamlit_app.py [hunk 1]
```diff
@@ -87,14 +99,14 @@ def architecture_dialog() -> None:
                 "Prompts, responses and feedback in this app are anonymously recorded and saved to LangSmith for product evaluation and improvement purposes only."
             )
 
+        st.markdown(f"Chat ID: **{st.session_state.thread_id}**")
```
Comment (JoshuaC215) [inline src/streamlit_app.py:102]: Add something like `help="Set ?thread_id=… to retrieve this conversation"`  so it’s more obvious?

Comment (peterkeppert) [reply src/streamlit_app.py]: Great idea, done!

**Comment by peterkeppert** [comment]
Right, I will try to create a service test.

As for more work needed, I was primarily thinking about the UI side of things, that it looks rather rudimentary. And also that maybe it is not so clear when we are continuing an existing chat and when we are having a new one and might want to save the Thread ID before closing the window.

**Comment by JoshuaC215** [comment]
Sweet - yeah from my side happy to merge after you add a test, and we can do further UI work in follow-on PRs. Whatever you prefer. (if the test gets too complex we could possibly delay that too)

**Comment by peterkeppert** [comment]
Service test added, please check.

**Review by JoshuaC215 — APPROVED** [review]
Nice work! Looks good!

**Pull request merged**